### PR TITLE
add option `retry` and `timeout`

### PR DESCRIPTION
**Pull request by BH4HPA**
In order to reduce the failure of API requests caused by network connection problems and the further suspension of GitHub Action tasks, two optional options `retry` and `timeout` are specially added.

The default value of the former `retry` is `3`, and the default value of the latter `timeout` is `10`. It is worth noting that the default value of `timeout` is `3` in Aliyun API, which is extended.
The test of the basic function has been passed, and further testing may be required, such as creating an environment with poor network connectivity to test the function of retrying.

### CODE REVIEW

**Comment by BH4HPA** [comment]
btw, by adding this feature #2 may be resolved.

File: index.js [hunk 1]
```diff
@@ -20,21 +22,34 @@ const input = {
...
-      accessKeyId: input.accessKeyId,
-      accessKeySecret: input.accessKeySecret
-    } : {},
-    ...input.securityToken ? {
-      securityToken: input.securityToken
-    } : {},
-    endpoint,
-    apiVersion
-  }).request(action, params, { method: "POST" });
+function callAliyunApi (endpoint, apiVersion, action, params) {
```
Comment (Menci) [inline index.js]: Please remove all the extra space between function name and left square bracket.

File: yarn.lock [hunk 2]
```diff
@@ -1,105 +1,152 @@
-# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
-# yarn lockfile v1
+# This file is generated by running "yarn install" inside your project.
```
Comment (Menci) [inline yarn.lock]: Don't overwrite the lockfile. I don't think our Yarn V1 support lockfile in this format.

**Review by Menci — CHANGES_REQUESTED** [review]

File: index.js [hunk 3]
```diff
@@ -20,21 +22,34 @@ const input = {
...
+      } : {},
+      ...input.securityToken ? {
+        securityToken: input.securityToken
+      } : {},
+      endpoint,
+      apiVersion
+    });
+
+    const request = () => client
+      .request(action, params, { method: "POST", timeout: input.timeout })
```
Comment (Menci) [inline index.js:41]: Convert `input.timeout` to `Number`?

**Review by Menci — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (BH4HPA) [reply index.js]: done

**Comment by Menci** [comment]
You should also update `action.yml` for adding inputs.

**Pull request merged**